diff --git a/dist/runtime/handler/node.js b/dist/runtime/handler/node.js
index 98e39da6e3a951ed9f49505aa3c014b1ebc6e16a..efc6c7d1c0f22f698d69788d726afd2040e46cb6 100644
--- a/dist/runtime/handler/node.js
+++ b/dist/runtime/handler/node.js
@@ -54,13 +54,14 @@ export const NodeHandler = opts => {
     const bundle = opts.bundle || {
         minify: true
     };
+    opts.artifactSrcPath = opts.srcPath.replaceAll("..", ".");
     // If srcPath is an absolute path, we need to convert it to an relative path
     // and append it to the artifact path.
     // Note: absolute "srcPath" should only be used for RDS's internal
     //       migrator function. User provided "srcPath" should always be
     //       relative path.
     const target = path.join(artifact, path
-        .relative(opts.root, path.resolve(opts.srcPath))
+        .relative(opts.root, path.resolve(opts.artifactSrcPath))
         .split(path.sep)
         .filter(x => x !== "node_modules")
         .join(path.sep), path.dirname(file), base + ".js");
@@ -182,7 +183,7 @@ export const NodeHandler = opts => {
             await runAfterBundling(opts.srcPath, artifact, bundle);
             const handler = path
                 .join(path
-                .relative(opts.root, path.resolve(opts.srcPath))
+                .relative(opts.root, path.resolve(opts.artifactSrcPath))
                 .split(path.sep)
                 .filter(x => x !== "node_modules")
                 .join(path.sep), opts.handler)